chrome/content/zotero/components/tagSelector/tagSelectorList.jsx

```diff
@@ -214,6 +221,70 @@ class TagList extends React.PureComponent {
 			</div>
 		);
 	};
+
+	shiftFocus(e) {
+		if (!document.activeElement.classList.contains("tag-selector-item")) return;
+		if (!["ArrowRight", "ArrowLeft"].includes(e.key)) return;
+		// Handle arrow navigation
+		let nextTag = (node) => {
+			if (e.key == "ArrowRight") return node.nextElementSibling;
+			return node.previousElementSibling;
+		};
+		let nextOne = nextTag(document.activeElement);
+		// Skip disabled tags
+		while (nextOne && nextOne.classList.contains("disabled")) {
+			nextOne = nextTag(nextOne);
+		}
+		if (nextOne) {
+			nextOne.focus();
+		}
+	}
+
+	handleSectionRendered = ({ indices }) => {
+		let tagsList = document.querySelector('.tag-selector-list');
+		// <Collection> sets role="grid" which is not semantically correct
+		tagsList.setAttribute("role", "group");
+
+		this.setState({ scrollToCell: undefined });
```

Comment:
I imagine this might not be necessary (and might cause an unnecessary rerender/call loop?), the element will only do something if scrollToCell is different from the previously provided one.

Reply:
Yeah this is a precaution to take care of a situation if the focused tag didn't actually change. For example, if you have the focus on the first tag, you scroll all the way down, press `arrowLeft` to scroll back to the first tag. After this, if you scroll down again and press `arrowLeft` again, we won't scroll back up because as far as the `<Collection>` is aware the `scrollToCell` prop has not changed. I suppose there can be more cases, especially if there are many disabled tags.

But now that I typed it, maybe it makes sense to just reset `scrollToCell` if the focused tag has not changed? I pushed that tweak, and I can't think of other cases when it would be required. That way, the edge case from  above is taken care of and we shouldn't re-render more often than needed